### ISSUE TITLE
Set-DbaTempDbConfiguration fails with "Split-Path : Cannot bind argument to parameter 'Path' because it is null", generates incorrect T-SQL statements

### ISSUE THREAD

**Issue by kozloski** [sqlcollaborative/dbatools#1363]
## Is this:
_Just place "x" to indicate selection (e.g. "[x]")_

- [ ] *feature* 
- [X] *bug*:

## System Details

- Operating system name and version: Windows 8.1 Enterprise
- Output from `$PSVersionTable`:

```
Name                           Value
----                           -----
PSVersion                      5.1.14409.1005
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.14409.1005
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

- Output of dbatools version:

```
Major  Minor  Build  Revision
-----  -----  -----  --------
0      8      957    -1
```

- SQL Server version for source/target
SQL Server 2014
SQL Server 2012

## Steps to Reproduce

1. New server with default tempdb configuration (1 data file, 1 log file.)
```
name     db_size       owner     dbid   created     status                                                                                                                                                                                                 compatibility_level
-------- ------------- --------- ------ ----------- ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ -------------------
tempdb        11.44 MB sa        2      May 30 2017 Status=ONLINE, Updateability=READ_WRITE, UserAccess=MULTI_USER, Recovery=SIMPLE, Version=782, Collation=SQL_Latin1_General_CP1_CI_AS, SQLSortOrder=52, IsAutoCreateStatistics, IsAutoUpdateStatistics  120

 
name     fileid filename                                        filegroup  size       maxsize    growth  usage
-------- ------ ----------------------------------------------- ---------- ---------- ---------- ------- ---------
tempdev  1      F:\MSSQL12.MSSQLSERVER\MSSQL\Data\tempdb.mdf    PRIMARY    10944 KB   Unlimited  10%     data only
templog  2      F:\MSSQL12.MSSQLSERVER\MSSQL\Data\templog.ldf   NULL       768 KB     Unlimited  10%     log only
```

2. Run Command:
```powershell
Set-DbaTempDbConfiguration -SqlInstance SERVERNAME -DataFileSizeMB 2048 -LogFileSizeMB 500 -WhatIf -Verbose
```
3. Received following error
```powershell
Split-Path : Cannot bind argument to parameter 'Path' because it is null.
At line:223 char:28
+                 $Filename = Split-Path $File.FileName -Leaf
+                                        ~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Split-Path], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorNullNotAllowed,Microsoft.PowerShell.Commands.SplitPathCo
   mmand

Split-Path : Cannot bind argument to parameter 'Path' because it is null.
At line:223 char:28
+                 $Filename = Split-Path $File.FileName -Leaf
+                                        ~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Split-Path], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorNullNotAllowed,Microsoft.PowerShell.Commands.SplitPathCo
   mmand
```

## Action Results
Set-DbaTempDbConfiguration fails to determine the logical file and physical file name for the existing tempdb data file. Generated statements are incorrect, with the logical file and physical file names missing, and three MODIFY FILE statements rather than two MODIFY FILE and one ADD FILE:

```
ALTER DATABASE tempdb MODIFY FILE(name=,filename='F:\MSSQL12.MSSQLSERVER\MSSQL\Data\',size=1024 MB,filegrowth=512);
ALTER DATABASE tempdb MODIFY FILE(name=,filename='F:\MSSQL12.MSSQLSERVER\MSSQL\Data\',size=1024 MB,filegrowth=512);
ALTER DATABASE tempdb MODIFY FILE(name=templog,filename='F:\MSSQL12.MSSQLSERVER\MSSQL\Data\templog.ldf',size=500 MB,filegrowth=512);
```
## Expected Results

1. New server with default tempdb configuration (1 data file, 1 log file.)
2. Run Command:
```powershell
Set-DbaTempDbConfiguration -SqlInstance SERVERNAME -DataFileSizeMB 2048 -LogFileSizeMB 500 -WhatIf -Verbose
```
3. Receive following output (if applicable)
No errors, generated statements have correct logical and physical file names.

```
ALTER DATABASE tempdb MODIFY FILE(name=tempdev,filename='F:\MSSQL12.MSSQLSERVER\MSSQL\Data\tempdb.mdf',size=1024 MB,filegrowth=512);
ALTER DATABASE tempdb MODIFY FILE(name=templog,filename='F:\MSSQL12.MSSQLSERVER\MSSQL\Data\templog.ldf',size=500 MB,filegrowth=512);
ALTER DATABASE tempdb ADD FILE (name=tempdev2,filename='F:\MSSQL12.MSSQLSERVER\MSSQL\Data\tempdev2.ndf',size=1024MB,filegrowth=512);
```
## Attached screen shots/console output

![image](https://cloud.githubusercontent.com/assets/5148846/26606217/b4b8be04-4556-11e7-8084-6d164a0ae8c4.png)

### Import - Attach dbatools logs

[dbatools_27876_message_0.log.txt](https://github.com/sqlcollaborative/dbatools/files/1039746/dbatools_27876_message_0.log.txt)

**Comment by kozloski** [comment]
I can repro this on a cow-orker's workstation as well, Windows 10 Enterprise with PowerShell 5.

This line appears to be the source of the issue:
```
$DataFiles = $server.Databases['tempdb'].ExecuteWithResults("select f.Name, f.physical_name as FileName from sys.filegroups fg join sys.database_files f on fg.data_space_id = fg.data_space_id where fg.name = 'PRIMARY' and f.type_desc = 'ROWS'").Tables
```
There are a couple of issues here:
1. The `$DataFiles` variable ends up being a collection of Tables object, rather than a single Table object. The quick fix for this would be `$DataFiles = (...).Tables[0]`.
2. This breaks on case-sensitive instances because `f.Name` can't be found. Should be `f.name as Name`

After looking over the code, I really think that the multiple calls to ExecuteWithResults could be replaced with pure SMO using the $server.Databases['tempdb'].FileGroups['PRIMARY].Files and $server.Databases['tempdb'].LogFiles collections. 

How far should I go with this? Fix the immediate issue now and come back with another PR for improvements?

**Comment by potatoqualitee** [comment]
Thanks for the issue report! Seems like one of my bugs. We'd love to have you fix it and PR to dev ASAP, @kozloski 👍 

The reason we don't use FileGroups is because it causes SMO Enumeration which slows down big time on larger systems.

What you could do is use a command in the dev branch `Get-DbaDatabaseFile -SqlInstance $server -Database tempdb`

The dev branch has massive changes but we're working through them.

**Comment by kozloski** [comment]
Fixed in #1375
